### PR TITLE
Alert when user unchecks `Active Application`

### PR DESCRIPTION
**Pull request by GoWind**
Part of the solution for Problem described [here](https://github.bus.zalan.do/pitchfork/issues/issues/16) (Internal link)

### CODE REVIEW

File: client/lib/application/src/application-form/application-form.jsx [hunk 1]
```diff
@@ -71,7 +71,19 @@ class ApplicationForm extends React.Component {
     /*eslint-disable react/no-direct-mutation-state */
     // TODO fix direct mutation of state here if possible
     update(field, prop, evt) {
-        if (field === 'criticality_level') {
+        
+        //Add an explicit warning if User unchecks the Active Application field
+        if (field === 'active') {
```
Comment (leviferreira) [inline client/lib/application/src/application-form/application-form.jsx:76]: There's a lot of code duplication with `this.state.app[field] = evt.target[prop];` + the if conditions are on 3 levels. Consider refactoring

File: client/lib/application/src/application-form/application-form.jsx [hunk 2]
```diff
@@ -71,7 +71,21 @@ class ApplicationForm extends React.Component {
...
+                let application_active = evt.target[prop];
+                let confirmed = false;
+                if (application_active === false) {
+                   confirmed = confirm('Deactivating Applications can cause Deployments to fail and or runtime failures. Please make sure you understand the risks of deactivating an application. Deactivate ?');
+                }
+
+                //Either application has been made active or (application was deactivated AND confirmed)
+                if(application_active || confirmed) {
```
Comment (hjacobs) [inline client/lib/application/src/application-form/application-form.jsx:84]: nit: formatting (space between `if` and `(`)

**Comment by leviferreira** [comment]
👍

**Comment by thilp** [comment]
:+1:

**Comment by GoWind** [comment]
:+1:

**Pull request merged**